common4j/src/main/com/microsoft/identity/common/java/util/HashMapExtensions.java

```diff
@@ -0,0 +1,158 @@
+
+    private HashMapExtensions() {
+        // Intentionally left blank
+    }
+
+    /**
+     * Decode url string into a key value pairs with default query delimiter.
+     *
+     * @param parameters URL query parameter
+     * @return key value pairs
+     */
+    public static HashMap<String, String> urlFormDecode(String parameters) {
+        return urlFormDecodeData(parameters, "&");
+    }
+
+    /**
+     * decode url string into a key value pairs with given query delimiter given
+     * string as a=1&b=2 will return key value of [[a,1],[b,2]].
+     *
+     * @param parameters URL parameter to be decoded
+     * @param delimiter  query delimiter
+     * @return Map key value pairs
+     */
+    static HashMap<String, String> urlFormDecodeData(String parameters, String delimiter) {
+        final String methodName = ":urlFormDecodeData";
+        final HashMap<String, String> result = new HashMap<>();
+
+        if (!StringUtil.isNullOrEmpty(parameters)) {
+            StringTokenizer parameterTokenizer = new StringTokenizer(parameters, delimiter);
+
+            while (parameterTokenizer.hasMoreTokens()) {
+                String pair = parameterTokenizer.nextToken();
+                String[] elements = pair.split("=");
+                String value = null;
+                String key = null;
+
+                if (elements.length == 2) {
+                    try {
+                        key = StringUtil.urlFormDecode(elements[0].trim());
+                        value = StringUtil.urlFormDecode(elements[1].trim());
+                    } catch (UnsupportedEncodingException e) {
+                        Logger.errorPII(
+                                TAG + methodName,
+                                "Encoding format is not supported",
+                                e
+                        );
+                        continue;
+                    }
+                } else if (elements.length == 1) {
+                    try {
+                        key = StringUtil.urlFormDecode(elements[0].trim());
+                        value = "";
+                    } catch (UnsupportedEncodingException e) {
+                        Logger.errorPII(
+                                TAG + methodName,
+                                "Encoding format is not supported",
+                                e
+                        );
+                        continue;
+                    }
+                }
+
+                if (!StringUtil.isNullOrEmpty(key)) {
+                    result.put(key, value);
+                }
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * get key value pairs from response.
+     *
+     * @param webResponse {@link HttpResponse} to convert to a map
+     * @return HashMap
+     * @throws JSONException
+     */
+    public static HashMap<String, String> getJsonResponse(HttpResponse webResponse) throws JSONException {
```

Comment:
Having these things return specifically HashMap is a bit strange.

Reply:
I'm removing some of the funcs in this class for now.. since we're not using them in this PR.

Including this one.

Reply:
actually.. it's the whole class. I think I might used it initially and scoped the size of the PR down.